packages/vector/valkey/cognee_community_vector_adapter_valkey/valkey_adapter.py

```
@@ -0,0 +1,509 @@
+        self._client: GlideClient | None = None
+        self._connected = False
+        self.VECTOR_DB_LOCK = asyncio.Lock()
+
+    # -------------------- lifecycle --------------------
+
+    async def get_connection(self) -> GlideClient:
+        """Establish and return an asynchronous Glide client connection to the Valkey server.
+
+        If a connection already exists and is marked as active, it will be reused.
+        Otherwise, a new connection is created using the configured host and port.
+
+        Returns:
+            GlideClient: An active Glide client instance for executing Valkey commands.
+
+        Behavior:
+            - Uses a backoff reconnect strategy with 3 retries and exponential delay.
+            - Disables TLS by default (set `use_tls=True` in configuration if needed).
+            - Sets a request timeout of 5000 ms.
+        """
+
+        if self._connected and self._client is not None:
+            return self._client
+
+        cfg = GlideClientConfiguration(
+            [NodeAddress(self._host, self._port)],
+            use_tls=False,
+            request_timeout=5000,
+            reconnect_strategy=BackoffStrategy(num_of_retries=3, factor=1000, exponent_base=2),
+        )
+        self._client = await GlideClient.create(cfg)
+        self._connected = True
+
+        return self._client
+
+    async def close(self) -> None:
+        """Close the active Glide client connection to the Valkey server.
+
+        If a client connection exists, attempts to close it gracefully.
+        Any exceptions during closure are suppressed to avoid breaking cleanup logic.
+
+        After closing:
+            - The internal client reference is set to None.
+            - The connection state flag (`_connected`) is reset to False.
+
+        Returns:
+            None
+
+        """
+
+        if self._client is not None:
+            try:
+                await self._client.close()
+            except Exception:
+                pass
```

Silent failures can be a nemesis.  Although this is a lower priority `close` call, consider logging any exception details here.